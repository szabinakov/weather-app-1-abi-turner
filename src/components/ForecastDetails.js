import React from "react";
import "../styles/ForecastDetails.css";

function ForecastDetails({ forecast }) {
  const { date, humidity, temperature, wind } = forecast;
  const formattedDate = new Date(date).toDateString();

  return (
    <div className="forecast-details" data-testid="forecast-details">
      <div className="forecast-details__date">{formattedDate}</div>
      <div className="forecast-details__temperature">
        Max Temperature: {temperature.max}
        &deg;C
      </div>
      <div className="forecast-details__humidity">Humidity:{humidity}%</div>
      <div className="forecast-details__wind-speed">
        Wind speed: {wind.speed}mph
      </div>
      <div className="forecast-details__wind-direction">
        Wind Direction: {wind.direction}
      </div>
    </div>
  );
}

export default ForecastDetails;
