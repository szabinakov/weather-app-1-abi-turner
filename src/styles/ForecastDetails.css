.forecast-details {

    margin-top: 5%;
    font-size: x-large;
    border-radius: 0px 0px 0px;
      background: silver;
    color: white;
    text-align: center;

}