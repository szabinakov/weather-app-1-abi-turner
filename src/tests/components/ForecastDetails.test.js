import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 1111111,
    temperature: {
      min: 12,
      max: 22,
    },
    humidity: 30,
    wind: {
      speed: 12,
      direction: "s",
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
