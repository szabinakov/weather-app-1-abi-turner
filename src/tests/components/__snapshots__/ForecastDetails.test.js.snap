// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
    data-testid="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Thu Jan 01 1970
    </div>
    <div
      class="forecast-details__temperature"
    >
      22Â°C
    </div>
    <div
      class="forecast-details__humidity"
    >
      30%
    </div>
    <div
      class="forecast-details__wind-speed"
    >
      12mph
    </div>
    <div
      class="forecast-details__wind-direction"
    >
      s
    </div>
  </div>
</DocumentFragment>
`;
